activerecord/lib/active_record/reflection.rb

```diff
@@ -117,7 +119,7 @@ def reflect_on_association(association)
       end
 
       def _reflect_on_association(association) # :nodoc:
-        _reflections[association.to_s]
+        _reflections[association.to_s] || _plural_reflections[association.to_s]
```

Comment:
This is a pull-down of an `||` behaviour that's previously lived in `_reflect_on_association`'s caller... it's nodoc, so we're free to redefine it in this way, but do we have any other in-framework callers that might be surprised by this? My recollection is that we call this a fair bit.

Reply:
Yeah, thinking about it more, I think having a new nodoc `_reflection_with_plurals` seems better than changing this behavior for all existing callers.